app/services/umrdr/date_form_service.rb

```diff
@@ -0,0 +1,58 @@
+module Umrdr
+  class DateFormService  
+
+    attr_reader :date_range
+    
+    def initialize(date_range)
+      @date_range = date_range
+    end  
+
+    def parse  
+
+      
+      if date_range.length>0
+     
+      	date_vals = []
+
+          #date_from_to = EDTF.parse(date_range) 
+          date_from_to = date_range.split('/')
+
+          start_dt = EDTF.parse(date_from_to[0])
+          end_dt =   EDTF.parse(date_from_to[1])
+     
+        if start_dt && start_dt.day_precision?
+          date_vals << start_dt.year 	
+          date_vals << start_dt.month 
+          date_vals << start_dt.day 
+        elsif start_dt && start_dt.month_precision?
+          date_vals << start_dt.year 	
+          date_vals << start_dt.month 
+          date_vals << nil 
+        elsif start_dt && start_dt.year_precision?
+          date_vals << start_dt.year 	
+          date_vals << nil
+          date_vals << nil  
+        end
+
+
+        if end_dt && end_dt.day_precision?
+          date_vals << end_dt.year 	
+          date_vals << end_dt.month 
+          date_vals << end_dt.day 
+        elsif end_dt && end_dt.month_precision?
+          date_vals << end_dt.year 	
+          date_vals << end_dt.month 
+          date_vals << nil 
+        elsif end_dt && end_dt.year_precision?
+          date_vals << end_dt.year 	
+          date_vals << nil
+          date_vals << nil  
+        end
```

Comment:
This block of logic is identical to the previous block of if - elsif logic.  Please extract to a function.